px-dropdown.html

```diff
@@ -129,478 +127,537 @@
+      this.set('selectedValues', []);
+      this.$.dropdown.close();
     },
     /**
-     * This function provide the cell size to the dropdown content.
+     * Forces the width of the dropdown to be at least as wide as the button.
      */
-    _provideCellSize: function (evt) {
-      var dropcell = this.contentAnchor ? this.contentAnchor : this.$.dropcell,
-        rect = dropcell.getBoundingClientRect();
-
-      evt.detail.pxContent.dropCellWidth = rect.width;
-      evt.detail.pxContent.dropCellHeight = rect.height;
-
-      evt.stopPropagation();
+    attached: function() {
+      this.$.dropdown.style.minWidth = window.getComputedStyle(this.$.button).width;
+      this._updateSelection();
     },
     /**
-     * This function checks whether the chevron should be visible or hidden.
+     * Calculates the class for a disabled dropdown.
      */
-    _hideChevron: function (newValue) {
-      return (!this.hideChevron);
+    _getDisabledClass: function() {
+      return this.disabled ? 'btn--disabled' : '';
     },
     /**
-     * Returns whether this dropdown is within the path.
+     * Calculates the class for a bare dropdown.
      */
-    _selfInPath: function (path) {
-      path = path || [];
-      for (var i = 0; i < path.length; i++) {
-        if (path[i] === this) {
-          return this;
-        }
-      }
+    _getClass: function() {
+      if(this.buttonStyle === 'bare') return 'btn--bare';
+      else if(this.buttonStyle === 'bare--primary') return 'btn--bare--primary';
+      else if(this.buttonStyle === 'tertiary') return 'btn--tertiary';
+      else return ''
     },
     /**
-     * Called whenever a click happens.
+     * Searches the DOM for the `boundTarget` element.
      */
-    _onCaptureClick: function (evt) {
-
-      //We're only interested in clicks:
-      // - not on ourselves
-      // - we're currently open
-      if (!this.preventCloseOnOutsideClick && !this._selfInPath(Polymer.dom(evt).path) &&
-        this.opened) {
-
-        var content = Polymer.dom(this).querySelector("px-dropdown-content");
-        this._flipOpened();
-        content.close();
-        this._reset();
-      }
+    _getElement: function(target) {
+      return document.querySelector(target);
     },
     /**
-     * This function is called when the bound target has changed.
+     * Any time that `items` changes, this method will convert an array
+     * of simple strings to an array of objects.
      */
-    _boundTargetChanged: function (boundTarget, isAttached) {
-      //find the element if we have been passed an id
-      if (typeof boundTarget === 'string') {
-        this.boundTarget = this.domHost ? this.domHost.$[boundTarget] :
-          Polymer.dom(this.ownerDocument).querySelector('#' + boundTarget);
+    _itemsChanged: function(items) {
+      if(items[0] && typeof items[0] === 'string') {
+        var length = items.length,
+            computedItems = [],
+            i;
+        for(i=0; i<length; i++) {
```

Comment:
Feel free to use the more semantically safe/easy to understand .forEach et al interators here, I doubt we're talking about large numbers of items here to perf is probably not a huge worry.